File: Mobile Programming/Mobile Programming/app/src/main/java/com/example/uasmobileprogramming/ApiNetwork.java
package com.example.uasmobileprogramming;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class ApiNetwork {
    public static final String base_url = "http://api.sharkhomedesign.com/public/";
    private static Retrofit retrofit = null;

    public static Retrofit getClient(){
        if (retrofit==null){
            retrofit= new Retrofit.Builder()
                    .baseUrl(base_url)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofit;
    }
}

File: app/src/main/java/com/example/belajar2/Remote/ApiNetwork.kt
package com.example.belajar2.Remote

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory


object ApiNetwork {
    private var retrofit: Retrofit? = null
    fun getClient(baseUrl: String) {
        if (retrofit == null) {
            retrofit = Retrofit.Builder()
                .baseUrl(baseUrl)
                .addConverterFactory(GsonConverterFactory.create())
                .build()
        }


        val base_url = "http://api.sharkhomedesign.com/public/"
    }
}


//   null val client: Retrofit?
//        get() {
//            if (retrofit == null) {
//                retrofit = Retrofit.Builder()
//                    .baseUrl(base_url)
//                    .addConverterFactory(GsonConverterFactory.create())
//                    .build()
//            }
//            return retrofit
//        }

    //    private var retrofit:Retrofit?=null
//    fun getclient(baseUrl:String):Retrofit{
//        if (retrofit==null)
//        {
//            retrofit= Retrofit.Builder()
//                .baseUrl(baseUrl)
//                .addConverterFactory(GsonConverterFactory.create())
//                .build()
//        }
//
//        return retrofit!!
//    }
